plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/viewer/handler/TreeViewerHandler.java

```diff
@@ -87,74 +56,91 @@ public TreeItem getTreeItem(Tree tree, String name) {
+	}
+	
+	private TreeItem getTreeItem(Iterator<TreeItem> iterator, String... path) {
+		List<TreeItem> resultItems = getTreeItems(iterator, path);
 
-		if (foundItems.size() == 0) {
-			throw new JFaceLayerException("There is no item with name " + name + " placed "
-				+ "under specified tree item");
-		}
-		
-		if (foundItems.size() > 1) {
-			throw new JFaceLayerException("Cannot choose specific tree item with name " + name 
-					+ " because there are more items with specified name"
-					+ " placed under specified tree item."); 
+		if (resultItems.size() > 1) {
+			throw new JFaceLayerException("There are more items matching specified path so result is ambiguous. " +
+					"To obtain all such items, please use method getTreeItems with same parameters");
 		}
 		
-		return foundItems.get(0);
+		return resultItems.get(0);
+	}	
+	
+	/**
+	 * Gets {@link TreeItem}s defined by a specified path placed under 
+	 * a specified {@link Tree}. If there are no suitable items under a 
+	 * specified tree, {@link JFaceLayerException} is thrown.
+	 * 
+	 * @param tree tree where to find tree items
+	 * @param path path to tree items
+	 * @return tree items placed under a specified tree
+	 */
+	public List<TreeItem> getTreeItems(Tree tree, String... path) {
+		return getTreeItems(tree.getItems().iterator(), path);
 	}
 	
 	/**
-	 * Gets tree item specified by path without decorators.
-	 * If there are more items with specified name then method throws JFaceLayerException.
-	 *
-	 * @param treeItem the tree item
-	 * @param path the path
-	 * @return the tree item
+	 * Gets {@link TreeItem}s defined by a specified path placed under 
+	 * a specified {@link TreeItem}. If there are no suitable items under a 
+	 * specified tree item, {@link JFaceLayerException} is thrown.
+	 * 
+	 * @param tree item tree item where to find tree items
+	 * @param path path to tree items
+	 * @return tree items placed under a specified tree item
 	 */
-	public TreeItem getTreeItem(TreeItem treeItem, String... path) {
-		TreeItem item = treeItem;
-		int index = 0;
+	public List<TreeItem> getTreeItems(TreeItem item, String... path) {
+		return getTreeItems(item.getItems().iterator(), path);
+	}
+	
+	private List<TreeItem> getTreeItems(Iterator<TreeItem> iterator, String... path) {
+		if (path.length == 0 || path == null) {
+			throw new IllegalArgumentException("Path to a tree item cannot be null");
+		}
 		
-		while (index < path.length - 1) {
-			item = getTreeItem(item, path[index]);
-			if (!item.isExpanded()) {
-				item.expand();
+		int index = 0;               
+		Iterator<TreeItem> nestedItemsIterator = iterator;
+		List<TreeItem> resultItems = new ArrayList<TreeItem>();
+	
+		while (index < path.length) {
+			List<TreeItem> nestedItems = new ArrayList<TreeItem>();
+			while (nestedItemsIterator.hasNext()) {
+				TreeItem item = nestedItemsIterator.next();
+				if (getNonStyledText(item).equals(path[index])) {
+					if (!item.isExpanded()) {
```

Comment:
you need to expand item only in else statement
if index==path.length -1 then we dont need to expand tree item. Expanding item is quite lengthy operation